libdevcrypto/OverlayDB.cpp

```diff
@@ -32,11 +32,11 @@ namespace dev
 
 h256 const EmptyTrie = sha3(rlp(""));
 
-OverlayDB::~OverlayDB()
-{
-	if (m_db.use_count() == 1 && m_db.get())
-		cnote << "Closing state DB";
-}
+#ifdef PRUNING
+u256 OverlayDB::m_blockNumber = 0;
+std::map<u256, std::set<h256> > OverlayDB::m_deathrow = {};
```

Comment:
The problem is, that every block creates a new instance. The deathrow and the m_changes need to be static because every object (state tree and storage tree for example) needs them. The m_blockNumber can be deleted later, since I only use it in the commit function. I put that in for debugging reasons, but will remove that latter.
Concerning order/unordered map, I can try the unordered. I did choose the ordered map because this set will always be used with successive keys. Don't know if that affect performance, will try.